src/main/java/in/ac/bits/protocolanalyzer/persistence/repository/SaveRepository.java

```diff
@@ -14,44 +16,49 @@
 import com.google.common.eventbus.EventBus;
 import com.google.common.eventbus.Subscribe;
 
-import javax.naming.Context;
-import javax.naming.InitialContext;
-import javax.naming.NamingException;
-
 import in.ac.bits.protocolanalyzer.analyzer.event.BucketLimitEvent;
 import in.ac.bits.protocolanalyzer.analyzer.event.EndAnalysisEvent;
 import in.ac.bits.protocolanalyzer.analyzer.event.SaveRepoEndEvent;
+import lombok.extern.log4j.Log4j;
 
 @Component
+@ComponentScan("config.in.ac.bits.protocolanalyzer.persistence.repository")
 @Scope("prototype")
 @Log4j
 public class SaveRepository implements Runnable {
 
 	@Autowired
 	private ElasticsearchTemplate template;
-
+	
+	@Autowired
 	private ConcurrentLinkedQueue<ArrayList<IndexQuery>> buckets;
-
+	
+	@Autowired
+	private Context ctx;
+	
+	@Autowired
+	private Runtime runtime;
+	
+	public long memory;
+	
 	private boolean isRunning = false;
 
-	private boolean analysisRunning = true;
+	public boolean analysisRunning = true;
 
 	private EventBus eventBus;
 
 	private int lowWaterMark;
 
 	private boolean analysisOnly;
-
+	
 	public boolean isRunning() {
 		return isRunning;
 	}
-
+	
 	public void configure(EventBus eventBus) {
-		buckets = new ConcurrentLinkedQueue<ArrayList<IndexQuery>>();
 		this.eventBus = eventBus;
 		this.eventBus.register(this);
 		try {
-			Context ctx = new InitialContext();
```

Comment:
Given that we are always using the context bean from `ctx.lookup("java:comp/env")`, we can directly return that bean from the config autowiring.

Reply:
Leads to #166

Reply:
@kapilkalra04 
In that case, can we pursue the autowiring of lowWaterMark and analysisOnly fields by putting the necessary autowiring logic in config class?

Reply:
So if we autowire lowWaterMark and AnalysisOnly using this bean creation code in the config file,
```
        @Bean
	public int lowWaterMark(){
		try {
			Context ctx = new InitialContext();
			Context env = (Context) ctx.lookup("java:comp/env");
			return Integer.parseInt((String) env.lookup("lowWaterMark"));
		} catch (NamingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		return 3;
	}
	
	@Bean
	public boolean analysisOnly(){
		try {
			Context ctx = new InitialContext();
			Context env = (Context) ctx.lookup("java:comp/env");
			return Boolean.parseBoolean((String) env.lookup("analysisOnly"));
		} catch (NamingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		return false;
	}
```
we can bypass creation the InitialContext() Beans, but in the SaveRepoTest class, we can't mock the new autowired fields, as they are primitive now. 

ExperimentTest will pass as well after adding the mentioned beans there as well in its config but with a warning of `javax.naming.NoInitialContextException`but the test passes and the application deploys correctly as well.

Reply:
So if we autowire lowWaterMark and AnalysisOnly using this bean creation code in the config file,
```
        @Bean
	public int lowWaterMark(){
		try {
			Context ctx = new InitialContext();
			Context env = (Context) ctx.lookup("java:comp/env");
			return Integer.parseInt((String) env.lookup("lowWaterMark"));
		} catch (NamingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		return 3;
	}
	
	@Bean
	public boolean analysisOnly(){
		try {
			Context ctx = new InitialContext();
			Context env = (Context) ctx.lookup("java:comp/env");
			return Boolean.parseBoolean((String) env.lookup("analysisOnly"));
		} catch (NamingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		return false;
	}
```
we can bypass creation the InitialContext() Beans, but in the SaveRepoTest class, we can't mock the new autowired fields, as they are primitive now. 

ExperimentTest will pass as well after adding the mentioned beans there as well in its config but with a warning of `javax.naming.NoInitialContextException`but the test passes and the application deploys correctly as well.